### PR TITLE
Update flag

### PR DESCRIPTION
**Pull request by jmccarty3**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+all: e2e
+
+dep:
+	dep ensure
+
+build: dep
+ifdef NEW_RELIC_APIKEY
 	git checkout deploy/operator.yaml
 	@sed -i '' s/REPLACE_ME_NEW_RELIC_APIKEY/${NEW_RELIC_APIKEY}/ deploy/operator.yaml
+	@sed -i '' s/REPLACE_NAMESPACE/default/ deploy/role_binding.yaml
```
Comment (sstarcher) [inline Makefile:16]: should we also checkout this file to reset it?

File: pkg/apis/newrelic/v1alpha1/monitor_types.go [hunk 2]
```diff
@@ -100,6 +101,11 @@ func (s *Monitor) IsCreated() bool {
 	return s.Status.IsCreated()
 }
 
+// Signature for the CRD
+func (s *Monitor) Signature() string {
```
Comment (sstarcher) [inline pkg/apis/newrelic/v1alpha1/monitor_types.go]: any reason you added this back?

File: pkg/apis/newrelic/v1alpha1/monitor_types.go [hunk 3]
```diff
@@ -215,21 +221,24 @@ func (s *Monitor) GetID() string {
...
-	s.Status.Hash = s.Spec.GetSum()
+		err = s.updateCondition(ctx)
+		if err != nil {
+			s.Status.Info = err.Error()
+			return err
+		}
+
+		s.Status.Hash = s.Spec.GetSum()
+	} else {
+		log.Warn(fmt.Sprintf("Skipping update for %s", s.Signature()))
```
Comment (sstarcher) [inline pkg/apis/newrelic/v1alpha1/monitor_types.go]: the signature should not be needed.  If you fetch the logger from the ctx.

**Review by sstarcher — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (jmccarty3) [reply Makefile:16]: Good point. Done

File: pkg/apis/newrelic/v1alpha1/monitor_types.go (hunk 2 above)
Comment (jmccarty3) [reply pkg/apis/newrelic/v1alpha1/monitor_types.go]: Did not know about the context logger. Removed.

File: pkg/apis/newrelic/v1alpha1/monitor_types.go (hunk 3 above)
Comment (jmccarty3) [reply pkg/apis/newrelic/v1alpha1/monitor_types.go]: Done.

**Pull request merged**